src/harness/fourslash.ts

```diff
@@ -3090,11 +3093,18 @@ ${code}
         let currentFileContent: string = undefined;
         let currentFileName = fileName;
         let currentFileOptions: { [s: string]: string } = {};
+        let currentFileSize = getFileSize();
+
+        function getFileSize() {
+            const stats = _fs.statSync("." + fileName);

```

Comment:
This should just be `this.content.length` - the file size you're reading from disk is the size of the fourslash test file that originated the file, but what you're asking for here is the size of the current source file in the in-memory representation of the file

Reply:
If the size is just `sourceFile.text.length`, this PR could be a lot simpler since it wouldn't require a new parameter.